src/bigint.hpp

```diff
@@ -18,6 +18,7 @@ struct BigInt {
         uint64_t *digits; // Least significant digit first
     } data;
     bool is_negative;
+    bool allow_binop_cast; //context information for casting ComptimeInt to Int
```

Comment:
This seems like a pretty big code smell. If this state needs to exist, this is almost certainly the wrong place for it. But I'm not convinced this state needs to exist.

Reply:
@andrewrk OK! I have a branch without `allow_binop_cast ` and hidden vars!